text/0000-run-uid.md

```diff
@@ -0,0 +1,59 @@
+# Meta
+[meta]: #meta
+- Name: Allowing different run-time and build-time uids
+- Start Date: 2021-03-15
+- Author(s): [@samj1912](https://github.com/samj1912)
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+[summary]: #summary
+
+This proposal proposes a change to the spec to allow for a different run-time user than the build-time user for various scenarios where you may want the run-time user to have different (ideally fewer) permissions than the build-time user.
+
+# Definitions
+[definitions]: #definitions
+
+Build image: An OCI image that provides the base of the build environment.
+
+Run image: An OCI image that provides the base from which app images are built.
+
+Build-time user: The user which is referenced in the `User` field of the build image.
+
+Run-time user: The user which is referenced in the `User` field of the run image.
+
+# Motivation
+[motivation]: #motivation
+
+- Why should we do this?
+
+Currently the [Platform Spec](https://github.com/buildpacks/spec/blob/main/platform.md#run-image) says that the run-time image's `User` along with its `CNB_USER_ID` environment variable must be set to the same user and uid as the build image.
+
+This prevents use cases where you may want the run-time user to not have the permissions to modify the built image during run-time and in general restrict the set of permissions on it. 
+
+The above restriction seems to impose limitations on such use cases which are valid and are supposed to be in line with the Cloud Native Buildpack policy of secure container environments (for eg. we prevent building/running as root user).
+
+This limitation seems to be enforced only by the spec and from my limited testing, it looks like platforms like [pack](https://github.com/buildpacks/pack) and [kpack](https://github.com/pivotal/kpack) don't actually enforce this and stacks with different run-time and build-time users seem to work as expected.
+
+- What is the expected outcome?
+
+Modification of the spec to allow a different run-time user than build-time. We can still enforce them being a part of the same unix group to avoid too much divergence from run-time and build-time images which may cause issues, while at the same time providing the flexibility to accommodate above use-cases.
```

Comment:
I wonder what constraint this translates to for Windows. Looking at https://github.com/buildpacks/rfcs/blob/main/text/0076-windows-security-identifiers.md I think it would mean requiring that the group SIDs are the same.

Reply:
Added some clarification on windows - does that look better?